File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
     public boolean isSubsequence(String s, String t) {
        String res = "";
        int i = 0;
        int y = 0;
        while (i < s.length() && y < t.length()) {
            if (s.charAt(i) == t.charAt(y)) {
                res += s.charAt(i);
                i++;
            }
            y++;
        }
        return res.equals(s);
    }
}

File: 0392-is-subsequence/0392-is-subsequence.kt
class Solution {
  fun isSubsequence(s: String, t: String): Boolean {
    var a = 0
    var b = 0
    if (s.isEmpty()) return true
    if (t.isEmpty()) return false
    while (b < t.length) {
        if (s[a] == t[b]) {
            a++
            b++
        } else {
            b++
        }
        if (a == s.length) return true
    }
    return false}

}